src/reflect/scala/reflect/internal/Definitions.scala

```diff
@@ -481,7 +481,7 @@ trait Definitions extends api.StandardDefinitions {
     // arrays and their members
     lazy val ArrayModule                   = requiredModule[scala.Array.type]
       lazy val ArrayModule_overloadedApply = getMemberMethod(ArrayModule, nme.apply)
-           def ArrayModule_genericApply    = ArrayModule_overloadedApply.suchThat(_.paramss.flatten.last.tpe.typeSymbol == ClassTagClass) // [T: ClassTag](xs: T*): Array[T]
+           def ArrayModule_genericApply    = ArrayModule_overloadedApply.suchThat(x => lastNonEmpty(x.paramss).last.tpe.typeSymbol == ClassTagClass) // [T: ClassTag](xs: T*): Array[T]
```

Comment:
Could we dodge this whole thing with
```scala
ArrayModule_overloadedApply.suchThat(_.paramss.length == 2)
```
?

It's not a very clean test but it does select the right method, I think. Alternatively, could we check if tparams is nonempty?

Reply:
You are probably right, however, I was aiming in this PR for quick, obvious, and mostly harmless hacks. I would rather not meddle with those assumptions for now.